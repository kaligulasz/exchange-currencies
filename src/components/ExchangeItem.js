import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

// Components
import MoneyInput from './MoneyInput';

const Title = styled.h3`
  font-size: 2.2rem;
  font-weight: 400;
  line-height: 1;
  margin: 0;
`;

const Wrapper = styled.section`
  box-sizing: border-box;
  height: 50%;
  ${({ secondary }) => secondary && `
    background: linear-gradient(to bottom, rgba(0,0,0,0.15) 0%,rgba(0,0,0,0) 90%);
  `};
`;

const Container = styled.div`
  margin: 0 auto;
  width: 70%;
  max-width: 500px;
  display: flex;
  height: 100%;
  align-items: center;
`;

const Item = styled.div`
  width: 100%;
`;

const Description = styled.div`
  opacity: 0.7;
  margin-top: 0.5rem;
`;

const ExchangeItem = ({
  pocket,
  primary,
  secondary,
  onAmountChange,
}) => {
  const { currency, currencySymbol, amount } = pocket;

  return (
    <Wrapper secondary={secondary}>
      <Container>
        <Item>
          <Title>{currency}</Title>
          <Description>
            You have
            <span>{currencySymbol}</span>
            {amount}
          </Description>
        </Item>

        {primary
          ? (
            <Item>
              <MoneyInput onChange={onAmountChange} />
            </Item>
          ) : (
            <div>{amount}</div>
          )
        }
      </Container>
    </Wrapper>
  );
};

ExchangeItem.defaultProps = {
  primary: false,
  secondary: false,
  onAmountChange: null,
};

ExchangeItem.propTypes = {
  pocket: PropTypes.object.isRequired,
  primary: PropTypes.bool,
  secondary: PropTypes.bool,
  onAmountChange: PropTypes.func,
};

export default ExchangeItem;
