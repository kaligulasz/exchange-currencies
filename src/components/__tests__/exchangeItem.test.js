import React from 'react';
import { shallow } from 'enzyme';

import ExchangeItem from '../ExchangeItem';

const mainPocket = {
  usd: {
    currencySymbol: '$',
    currency: 'usd',
    description: 'USD - American Dollar',
    amount: 30.31,
    id: 1,
  },
  eur: {
    currencySymbol: '€',
    currency: 'eur',
    description: 'EUR - Euro',
    amount: 20.1,
    id: 2,
  },
  gbp: {
    currencySymbol: '£',
    currency: 'gbp',
    description: 'GBP - British Pound',
    amount: 25.55,
    id: 3,
  },
};

const pocket = {
  currencySymbol: '$',
  currency: 'usd',
  description: 'USD - American Dollar',
  amount: 30.31,
  id: 1,
};

describe('<ExchangeItem />', () => {
  it('renders <ExchangeItem primary/>', () => {
    const component = shallow(
      <ExchangeItem
        mainPocket={mainPocket}
        pocket={pocket}
        onAmountChange={() => {}}
        primary
      />,
    );

    expect(component).toMatchSnapshot();
  });

  it('renders <ExchangeItem secondary/>', () => {
    const component = shallow(
      <ExchangeItem
        mainPocket={mainPocket}
        pocket={pocket}
        actualRate={0.79}
        changingFromCurrencySymbol="$"
        secondary
      />,
    );

    expect(component).toMatchSnapshot();
  });
});
