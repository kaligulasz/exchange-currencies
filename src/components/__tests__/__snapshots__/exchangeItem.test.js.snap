// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExchangeItem /> renders <ExchangeItem primary/> 1`] = `
<ExchangeItem__Wrapper
  secondary={false}
>
  <ExchangeItem__Container>
    <ExchangeItem__Item>
      <ExchangeItem__Title>
        usd
      </ExchangeItem__Title>
      <ExchangeItem__Description>
        You have $30.31
      </ExchangeItem__Description>
    </ExchangeItem__Item>
    <ExchangeItem__Item>
      <MoneyInput
        maxValue={30.31}
        onChange={[Function]}
      />
    </ExchangeItem__Item>
  </ExchangeItem__Container>
</ExchangeItem__Wrapper>
`;

exports[`<ExchangeItem /> renders <ExchangeItem secondary/> 1`] = `
<ExchangeItem__Wrapper
  secondary={true}
>
  <ExchangeItem__Container>
    <ExchangeItem__Item>
      <ExchangeItem__Title>
        usd
      </ExchangeItem__Title>
      <ExchangeItem__Description>
        You have $30.31
      </ExchangeItem__Description>
    </ExchangeItem__Item>
    <ExchangeItem__ResultWrapper>
      <ExchangeItem__Result>
        <Plus
          size="20"
        />
        <ExchangeItem__Title>
          0
        </ExchangeItem__Title>
      </ExchangeItem__Result>
      <ExchangeItem__Description
        textAlignRight={true}
      >
        $1 = $0.79
      </ExchangeItem__Description>
    </ExchangeItem__ResultWrapper>
  </ExchangeItem__Container>
</ExchangeItem__Wrapper>
`;
